### PR TITLE
The "Host" header is being preserved, if it was set manually.

### PR DESCRIPTION
**Pull request by lexakozakov**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "guzzlehttp/guzzle",
+    "name": "lexakozakov/guzzle",
```
Comment (SofHad) [inline composer.json]: you must not commit this.

Comment (lexakozakov) [inline composer.json]: My fault

File: src/Client.php [hunk 2]
```diff
@@ -93,7 +93,7 @@ public function sendAsync(RequestInterface $request, array $options = [])
         $options = $this->prepareDefaults($options);
 
         return $this->transfer(
-            $request->withUri($this->buildUri($request->getUri(), $options)),
+            $request->withUri($this->buildUri($request->getUri(), $options), $request->hasHeader("Host")),
```
Comment (SofHad) [inline src/Client.php]: Single quotes for consstency.

File: tests/ClientTest.php [hunk 3]
```diff
@@ -592,4 +592,27 @@ public function testProperlyBuildsQuery()
         $client->send($request, ['query' => ['foo' => 'bar', 'john' => 'doe']]);
         $this->assertEquals('foo=bar&john=doe', $mock->getLastRequest()->getUri()->getQuery());
     }
+
+    public function testSendSendsWithIpAddressAndPortAndHostHeaderInRequestTheHostShouldBePreserved()
+    {
+        $mockHandler = new MockHandler([new Response(200)]);
+        $client = new Client(['base_uri' => '127.0.0.1:8585', 'handler' => $mockHandler]);
+        $request = new Request('GET', '/test', ["Host"=>"foo.com"]);
```
Comment (SofHad) [inline tests/ClientTest.php]: Single quotes for consistency.

File: tests/ClientTest.php [hunk 4]
```diff
@@ -592,4 +592,27 @@ public function testProperlyBuildsQuery()
...
+        $client->send($request);
+
+        $this->assertEquals("foo.com", $mockHandler->getLastRequest()->getHeader("Host")[0]);
+    }
+
+    public function testSendSendsWithDomainAndHostHeaderInRequestTheHostShouldBePreserved()
+    {
+        $mockHandler = new MockHandler([new Response(200)]);
+        $client = new Client(['base_uri' => 'http://foo2.com', 'handler' => $mockHandler]);
+        $request = new Request('GET', '/test', ["Host"=>"foo.com"]);
```
Comment (SofHad) [inline tests/ClientTest.php]: Single quotes for consistency.

File: tests/ClientTest.php [hunk 5]
```diff
@@ -592,4 +592,27 @@ public function testProperlyBuildsQuery()
...
+
+        $client->send($request);
+
+        $this->assertEquals("foo.com", $mockHandler->getLastRequest()->getHeader("Host")[0]);
```
Comment (SofHad) [inline tests/ClientTest.php]: the same

File: src/Client.php (hunk 2 above)
Comment (lexakozakov) [inline src/Client.php]: I've created another [pull request](https://github.com/guzzle/guzzle/pull/1376).

**Pull request closed without merging**